app/controllers/v0/institutions_controller.rb

```diff
@@ -114,17 +114,17 @@ def search_results
     # rubocop:enable Metrics/MethodLength
 
     # TODO: If filter counts are desired in the future, change boolean facets
-    # to use search_results.filter_count(param) instead of default value
+    # to use count_field(search_results, :field_name) instead of default value
     def facets
-      institution_types = search_results.filter_count(:institution_type_name)
+      institution_types = count_field(search_results, :institution_type_name)
       result = {
         category: {
           school: institution_types.except(Institution::EMPLOYER).inject(0) { |count, (_t, n)| count + n },
           employer: institution_types[Institution::EMPLOYER].to_i
         },
         type: institution_types,
-        state: search_results.filter_count(:state),
-        country: embed(search_results.filter_count(:country)),
+        state: count_field(search_results, :state),
+        country: embed(count_field(search_results, :country)),
```

Comment:
Did you test (measure) these changes?
This change makes requests much slower, depending on the number of results.
If I visit http://localhost:3000/v0/institutions (no search terms) on my machine, the requests are 5x - 10x SLOWER.
That's worst case scenario, but the more results terms the worse it gets.
These changes do not improve speed at any search results threshold.

Here are two examples each of the new way and old way when I visit http://localhost:3000/v0/institutions:

Old results:
Completed 200 OK in **902ms** (Views: 171.5ms | ActiveRecord: 668.0ms)
Completed 200 OK in **419ms** (Views: 26.8ms | ActiveRecord: 382.0ms)

New Results:
Completed 200 OK in **5002ms** (Views: 38.5ms | ActiveRecord: 541.9ms)
Completed 200 OK in **4581ms** (Views: 41.2ms | ActiveRecord: 678.4ms)

You removed a SQL count statement and replaced it with a Ruby function that has to instantiate every search result object and loop through them (not just results for the current page; _all_ matching results).

SQL is (generally) faster than Ruby.  We need to optimize our SQL, and remove SQL if we can, but replacing SQL with Ruby is almost never the right answer.

Reply:
Yes, I tested the performance, but with specific search terms and got much smaller result sets and got much different results.I don't think we have a route that would allow searching without any terms, but I am seeing the performance issues with terms that return huge result sets ("name=university" unsurprisingly has a large amount of records).